### PR TITLE
Log LR using LearningRateMonitor even when LR Scheduler is not defined. 

### LINKED ISSUE #7468: Extend the Learning Rate Monitor callback to log the initial lr for optimizers
## 🚀 Feature
Log the initial learning rate of each optimizer

### Motivation

Currently the LR monitor callback only takes effect if the user specifies LR schedulers for training. However, one could want to record what the initial learning rate used for their optimizer's parameter groups even without any LR schedulers. 

### Pitch

We could log this once on train start following the logic here: https://github.com/PyTorchLightning/pytorch-lightning/blob/f6fe715e73cc87bfc106329f9103564097515379/pytorch_lightning/callbacks/lr_monitor.py#L138-L156

cc @simran2905

### PR DESCRIPTION
**Pull request by VirajBagal**
## What does this PR do?
In this PR, the `LearningRateMonitor` callback is extended to log learning rates of each optimizer even when LR schedulers are not specified by the user in training. 

When LR schedulers are not specified, `trainer.optimizers` is used instead of `trainer.lr_schedulers` for getting names of the optimizers, and then we iterate over the optimizers to log their learning rates.  

Extra tests are written to make sure that the learning rates are logged with correct names even when schedulers are not defined. 

## Motivation
Currently the LR monitor callback only takes effect if the user specifies LR schedulers for training. However, one could want to record what the initial learning rate used for their optimizer's parameter groups even without any LR schedulers. This PR tries to solve this issue. 
Fixes #7468 

### Does your PR introduce any breaking changes? 
No

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: pytorch_lightning/callbacks/lr_monitor.py [hunk 1]
```diff
@@ -155,12 +154,30 @@ def on_train_epoch_start(self, trainer, *args, **kwargs):
...
+                    use_betas = "betas" in opt.defaults
+
+                    for i, pg in enumerate(param_groups):
+                        name_and_suffix = self._add_suffix(name, param_groups, i)
+                        lr = self._extract_lr(pg, name_and_suffix)
+                        latest_stat.update(lr)
+                        momentum = self._extract_momentum(
+                            param_group=pg, name=name_and_suffix.replace(name, f"{name}-momentum"), use_betas=use_betas
+                        )
+                        latest_stat.update(momentum)
```
Comment (ananthsub) [inline pytorch_lightning/callbacks/lr_monitor.py:174]: this code is duplicated in the else case below. could we abstract this to a common helper function to reduce the risk of divergence?

File: pytorch_lightning/callbacks/lr_monitor.py [hunk 2]
```diff
@@ -259,6 +276,35 @@ def _find_names(self, lr_schedulers: List, add_lr_sch_names: bool = True) -> Lis
...
+                    f"`name` values. {name} has duplicated parameter group names {duplicates}"
+                )
+
+            name = self._add_prefix(name, optimizer_cls, seen_optimizer_types)
+
+            names.extend(self._add_suffix(name, param_groups, i) for i in range(len(param_groups)))
+
+            if add_lr_sch_names:
+                self.lr_sch_names.append(name)
+
```
Comment (ananthsub) [inline pytorch_lightning/callbacks/lr_monitor.py:305]: same here. there's a lot of duplicated code above. let's try to abstract these into common helper functions

Comment (VirajBagal) [reply pytorch_lightning/callbacks/lr_monitor.py]: I have added helper methods.

File: pytorch_lightning/callbacks/lr_monitor.py (hunk 1 above)
Comment (VirajBagal) [reply pytorch_lightning/callbacks/lr_monitor.py]: I have now added helper methods
